client/battle/BattleInterface.cpp

```diff
@@ -97,7 +97,17 @@ BattleInterface::BattleInterface(const CCreatureSet *army1, const CCreatureSet *
 
 	CCS->musich->stopMusic();
 	setAnimationCondition(EAnimationEvents::OPENING, true);
-	battleIntroSoundChannel = CCS->soundh->playSoundFromSet(CCS->soundh->battleIntroSounds);
+
+	GH.pushInt(windowObject);
+	windowObject->blockUI(true);
+	windowObject->updateQueue();
+
+	if(settings["gameTweaks"]["skipBattleIntroMusic"].Bool())

```

Comment:
Maybe move everything from this point and to the end of method into a separate method, e.g. playIntroSound?

Reply:
done